Mathlib/RingTheory/Polynomial/Eisenstein/Distinguished.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2024 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan
+-/
+
+import Mathlib.RingTheory.Polynomial.Eisenstein.Basic
+import Mathlib.RingTheory.PowerSeries.Order
+/-!
+
+# Distiguished polynomial
+
+In this file we define the structure `IsDistinguishedAt`
+and develop the most basic lemmas about it.
+
+-/
+
+open scoped Polynomial
+open PowerSeries Ideal Quotient
+
+variable {R : Type*} [CommRing R]
+
+/-- Given an ideal `I` of a commutative ring `R`, we say that a polynomial `f : R[X]`
+is *Distinguished at `I`* if `f` is monic and `IsWeaklyEisensteinAt I`. -/
+structure Polynomial.IsDistinguishedAt (f : R[X]) (I : Ideal R) : Prop
```

Comment:
Since `IsWeaklyEisensteinAt` is AFAIK not standard terminology in the literature, maybe remind the user what this means under the hood?
(e.g. "i.e. `f` is of the form `xⁿ + a₁xⁿ⁻¹ + ⋯ + aₙ` with `aᵢ ∈ I` for all `i`")